### PR TITLE
Add vectorEmbeddingPolicy support to Spark catalog container creation

### PR DESCRIPTION
**Pull request by xinlian12**
## Summary

Enable configuring **Vector Embedding Policy** via `TBLPROPERTIES` when creating containers through the Spark catalog API. This allows users to set up vector-search-enabled containers directly from Spark SQL.

## Changes

- **`CosmosContainerProperties.scala`** — Added `vectorEmbeddingPolicy` TBLPROPERTIES key and `getVectorEmbeddingPolicy()` getter
- **`SparkModelBridgeInternal.scala`** — Added `createVectorEmbeddingPolicyFromJson()` and `vectorEmbeddingPolicyToJson()` bridge methods
- **`CosmosCatalogCosmosSDKClient.scala`** — Applied vector embedding policy in `createContainer()` and exposed it in `generateTblProperties()` metadata
- **`CosmosCatalogManagementSDKClient.scala`** — Same changes for the management SDK client path
- **`CosmosConstants.scala`** — Added `VectorEmbeddingPolicy` constant to `TableProperties`
- **`CosmosCatalogITestBase.scala`** — Added integration test; updated all `tblProperties` size assertions from 7→8

## Usage Example

```sql
CREATE TABLE catalog.db.container (word STRING, number INT) USING cosmos.oltp
  TBLPROPERTIES(
    partitionKeyPath = '/mypk',
    manualThroughput = '1100',
    indexingPolicy = '{"indexingMode":"consistent","automatic":true,"includedPaths":[{"path":"/*"}],"excludedPaths":[{"path":"/_etag/?"}],"vectorIndexes":[{"path":"/vector1","type":"flat"}]}',
    vectorEmbeddingPolicy = '{"vectorEmbeddings":[{"path":"/vector1","dataType":"float32","distanceFunction":"cosine","dimensions":1536}]}'
  )
```

## Verification

- All source and test code compiles successfully
- Integration test validates: container creation with vector embedding policy, vector indexes in indexing policy, metadata readback of the policy

### Currently Supported

| Property | Notes |
|---|---|
| Partition Key Definition | Core property |
| Indexing Policy | JSON config via `indexingPolicy` |
| Default TTL | Integer seconds via `defaultTtlInSeconds` |
| Analytical Store TTL | Integer seconds via `analyticalStoreTtlInSeconds` |
| Vector Embedding Policy | **Added in this PR** via `vectorEmbeddingPolicy` |

### Not Yet Supported (https://github.com/Azure/azure-sdk-for-java/issues/48360)

| Property | SDK Class |
|---|---|
| Unique Key Policy | `UniqueKeyPolicy` |
| Conflict Resolution Policy | `ConflictResolutionPolicy` |
| Computed Properties | `ComputedProperty` |
| Client Encryption Policy | `ClientEncryptionPolicy` |
| Full Text Policy | `CosmosFullTextPolicy` |
| Change Feed Policy | `ChangeFeedPolicy` |

### CODE REVIEW

**Comment by xinlian12** [comment]
## 🔍 Deep Review — PR #48349: Add vectorEmbeddingPolicy support to Spark catalog

**Summary:** This PR adds `vectorEmbeddingPolicy` as a new `TBLPROPERTIES` key for Spark catalog container creation, enabling vector-search-enabled Cosmos DB containers from Spark SQL. The implementation follows existing patterns closely across both SDK client paths (Cosmos SDK + Management SDK), adds JSON bridge methods, and includes an integration test.

**Overall Assessment:** The code is well-structured and follows existing conventions. However, there is one blocking issue (pre-existing but amplified by this PR) where the management SDK path silently drops `vectorIndexes` from the indexing policy, rendering vector search non-functional through the AAD auth path. Several test coverage gaps and minor improvements are also noted.

### Findings Summary

| # | Severity | Category | File | Finding |
|---|----------|----------|------|---------|
| 1 | 🔴 Blocking | Correctness | `CosmosCatalogManagementSDKClient.scala` | Management SDK drops `vectorIndexes` from indexing policy |
| 2 | 🟡 Recommendation | Test Coverage | `CosmosCatalogITestBase.scala` | No test for management SDK path |
| 3 | 🟡 Recommendation | Maintainability | `CosmosCatalogManagementSDKClient.scala` | Inconsistent serialization pattern between clients |
| 4 | 🟡 Recommendation | Test Coverage | `CosmosCatalogITestBase.scala` | Existing tests don't assert VEP = "null" |
| 5 | 🟡 Recommendation | Reliability | `CosmosContainerProperties.scala` | No cross-validation between VEP and vectorIndexes |
| 6 | 🟡 Recommendation | Reliability | `SparkModelBridgeInternal.scala` | No error handling for malformed JSON |
| 7 | 🟢 Suggestion | Test Quality | `CosmosCatalogITestBase.scala` | Weak `.contains()` assertions |
| 8 | 🟢 Suggestion | Edge Case | `CosmosCatalogCosmosSDKClient.scala` | Empty VEP serialization |
| 9 | 🟢 Suggestion | Observability | `CosmosCatalogCosmosSDKClient.scala` | No logging when applying VEP |

💬 **Observation:** Branch name `allowConfigVectorSearchIndexInSparkCataglog` contains a typo — "Cataglog" → "Catalog".

See inline comments for details on each finding.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala [hunk 1]
```diff
@@ -116,6 +116,14 @@ private[spark] case class CosmosCatalogManagementSDKClient(resourceGroupName: St
             case None =>
         }
 
+        // setup vector embedding policy
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala:119]: **🔴 Blocking** · Correctness: Silent Data Loss `[Already posted by @Copilot — Unresolved]`

**Management SDK path silently drops `vectorIndexes` from indexing policy, making vector search non-functional**

The `getIndexingPolicy()` method (lines ~240-270) only maps `includedPaths`, `excludedPaths`, and `indexingMode` from the data-plane `IndexingPolicy` to the management SDK `IndexingPolicy`. It **silently drops** `vectorIndexes`, `compositeIndexes`, and `spatialIndexes`.

When a user sets both `vectorEmbeddingPolicy` and `indexingPolicy` with `vectorIndexes` through the management SDK path (AAD auth), the container will have the embedding policy but **no vector indexes** — making vector search non-functional. This is a pre-existing bug, but this PR makes it user-visible by enabling vector embedding policy configuration.

Copilot flagged this same issue; no response from the author yet. Consider extending `getIndexingPolicy()` to also map vector indexes.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala [hunk 2]
```diff
@@ -480,6 +480,73 @@ abstract class CosmosCatalogITestBase(val skipHive: Boolean = false) extends Int
     tblProperties("DefaultTtlInSeconds") shouldEqual "5"
   }
 
+  it can "create a table with vector embedding policy" in {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala:487]: **🟡 Recommendation** · Reliability: Test Coverage

**No test coverage for the management SDK client path**

This integration test only exercises the **Cosmos SDK client** path. The management SDK client (`CosmosCatalogManagementSDKClient`) has fundamentally different serialization logic — it uses `objectMapper.readValue(json, classOf[MgmtVectorEmbeddingPolicy])` with the `JsonSerializableDatabindModule` bridge, a different deserialization path from the data-plane Jackson `@JsonProperty` annotations.

Without a test, there is no proof the user-provided JSON correctly deserializes into `com.azure.resourcemanager.cosmos.models.VectorEmbeddingPolicy` via the Jackson→`JsonSerializable` bridge. The management SDK path is used for AAD auth scenarios.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala [hunk 3]
```diff
@@ -116,6 +116,14 @@ private[spark] case class CosmosCatalogManagementSDKClient(resourceGroupName: St
...
+        CosmosContainerProperties.getVectorEmbeddingPolicy(containerProperties) match {
+            case Some(vectorEmbeddingPolicyJson) =>
+                sqlContainerResource.withVectorEmbeddingPolicy(
+                    objectMapper.readValue(vectorEmbeddingPolicyJson, classOf[MgmtVectorEmbeddingPolicy]))
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala]: **🟡 Recommendation** · Maintainability: Inconsistent Patterns

**Serialization pattern diverges from the Cosmos SDK client**

The Cosmos SDK client delegates VEP serialization to `SparkModelBridgeInternal.createVectorEmbeddingPolicyFromJson()` and `vectorEmbeddingPolicyToJson()`, centralizing serialization. This management client does inline `objectMapper.readValue()` directly.

This asymmetry means:
- Future JSON format changes need to be updated in two places
- The two paths could produce subtly different JSON during readback (`generateTblProperties`)
- The `objectMapper` here includes `JsonSerializableDatabindModule` while `SparkModelBridgeInternal` uses `Utils.getSimpleObjectMapper` — different serialization capabilities

Consider adding bridge methods for the management type, or documenting why the asymmetry is intentional.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala [hunk 4]
```diff
@@ -158,7 +158,7 @@ abstract class CosmosCatalogITestBase(val skipHive: Boolean = false) extends Int
 
     val tblProperties = getTblProperties(spark, databaseName, containerName)
 
-    tblProperties should have size 7
+    tblProperties should have size 8
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala:161]: **🟡 Recommendation** · Reliability: Test Gap

**Existing tests updated to size 8 but missing `VectorEmbeddingPolicy` assertion**

All six existing tests were updated from `have size 7` to `have size 8`, but none add an assertion like:

```scala
tblProperties("VectorEmbeddingPolicy") shouldEqual "null"
```

This is inconsistent with how these same tests already assert `tblProperties("AnalyticalStoreTtlInSeconds") shouldEqual "null"` for containers without analytical store TTL. Without this assertion, a bug in the null-path (e.g., returning an empty JSON object instead of `"null"`) would go undetected.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosContainerProperties.scala [hunk 5]
```diff
@@ -45,4 +46,8 @@ private[spark] object CosmosContainerProperties {
       None
     }
   }
+
+  def getVectorEmbeddingPolicy(properties: Map[String, String]): Option[String] = {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosContainerProperties.scala:50]: **🟡 Recommendation** · Reliability: Missing Cross-Validation

**No validation that `vectorEmbeddingPolicy` and `indexingPolicy` are consistent**

Vector search requires coordination between two properties: `vectorEmbeddingPolicy` defines embedding paths/dimensions, and `indexingPolicy.vectorIndexes` defines how those paths are indexed.

A user could provide `vectorEmbeddingPolicy` without corresponding `vectorIndexes` in `indexingPolicy` (or vice versa), or with mismatched paths. This silently creates a non-functional vector search configuration.

At minimum, consider logging a warning when `vectorEmbeddingPolicy` is provided but no `vectorIndexes` are found in `indexingPolicy`.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/models/SparkModelBridgeInternal.scala [hunk 6]
```diff
@@ -3,12 +3,24 @@
...
+
   def createIndexingPolicyFromJson(json: String): IndexingPolicy = {
     new IndexingPolicy(json)
   }
 
   def createPartitionKeyDefinitionFromJson(json: String): PartitionKeyDefinition = {
     new PartitionKeyDefinition(json)
   }
+
+  def createVectorEmbeddingPolicyFromJson(json: String): CosmosVectorEmbeddingPolicy = {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/models/SparkModelBridgeInternal.scala:19]: **🟡 Recommendation** · Reliability: Error Handling

**Malformed JSON produces a cryptic Jackson stack trace**

`objectMapper.readValue(json, classOf[CosmosVectorEmbeddingPolicy])` will throw a raw `JsonProcessingException` if the user provides malformed JSON in `TBLPROPERTIES`. Spark SQL users will see a cryptic Jackson deserialization error rather than an actionable message.

The same issue applies to the management client path (inline `objectMapper.readValue`). Consider wrapping the deserialization with a try-catch that re-throws as an `IllegalArgumentException` with a user-friendly message explaining the expected JSON format.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala [hunk 7]
```diff
@@ -480,6 +480,73 @@ abstract class CosmosCatalogITestBase(val skipHive: Boolean = false) extends Int
...
+    val tblProperties = getTblProperties(spark, databaseName, containerName)
+
+    tblProperties should have size 8
+
+    tblProperties("CosmosPartitionKeyDefinition") shouldEqual "{\"paths\":[\"/mypk\"],\"kind\":\"Hash\"}"
+    tblProperties("DefaultTtlInSeconds") shouldEqual "null"
+    tblProperties("AnalyticalStoreTtlInSeconds") shouldEqual "null"
+
+    // validate vector embedding policy is in table properties
+    tblProperties("VectorEmbeddingPolicy").contains("vector1") shouldEqual true
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala]: **🟢 Suggestion** · Reliability: Test Quality

**Substring checks could pass with broken JSON structure**

These `.contains()` assertions would pass even if the JSON structure is malformed (e.g., the key `"path"` is missing but `"vector1"` appears elsewhere in the string).

A stronger assertion would parse the JSON and check structured fields, matching the approach used above for the container properties validation (which checks `embedding.getPath`, `embedding.getDataType`, etc.):

```scala
val vepJson = tblProperties("VectorEmbeddingPolicy")
val vep = SparkModelBridgeInternal.createVectorEmbeddingPolicyFromJson(vepJson)
vep.getVectorEmbeddings should have size 1
vep.getVectorEmbeddings.get(0).getPath shouldEqual "/vector1"
```

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala [hunk 8]
```diff
@@ -299,6 +306,11 @@ private[spark] case class CosmosCatalogCosmosSDKClient(cosmosAsyncClient: Cosmos
             case None => "null"
         }
 
+        val vectorEmbeddingPolicySnapshot = Option.apply(containerProperties.getVectorEmbeddingPolicy) match {
+            case Some(policy) => SparkModelBridgeInternal.vectorEmbeddingPolicyToJson(policy)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala]: **🟢 Suggestion** · Correctness: Edge Case

**Non-null but empty `CosmosVectorEmbeddingPolicy` could serialize inconsistently**

If `getVectorEmbeddingPolicy` returns a non-null `CosmosVectorEmbeddingPolicy` with an empty `vectorEmbeddings` list, this serializes as `{"vectorEmbeddings":[]}` rather than `"null"`, inconsistent with other properties that show `"null"` when not meaningfully set.

Consider guarding against empty policies:

```scala
val vectorEmbeddingPolicySnapshot = Option.apply(containerProperties.getVectorEmbeddingPolicy)
    .filter(p => p.getVectorEmbeddings != null && !p.getVectorEmbeddings.isEmpty) match {
        case Some(policy) => SparkModelBridgeInternal.vectorEmbeddingPolicyToJson(policy)
        case None => "null"
    }
```

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala [hunk 9]
```diff
@@ -96,6 +96,13 @@ private[spark] case class CosmosCatalogCosmosSDKClient(cosmosAsyncClient: Cosmos
             case None =>
         }
 
+        CatalogCosmosContainerProperties.getVectorEmbeddingPolicy(containerProperties) match {
+            case Some(vectorEmbeddingPolicyJson) =>
+                cosmosContainerProperties.setVectorEmbeddingPolicy(
+                    SparkModelBridgeInternal.createVectorEmbeddingPolicyFromJson(vectorEmbeddingPolicyJson))
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala:102]: **🟢 Suggestion** · Maintainability: Observability

**No logging when vector embedding policy is applied**

Neither SDK client logs when the vector embedding policy is applied during container creation. For a new feature that requires coordination with indexing policy (`vectorIndexes`), a diagnostic log message would help with troubleshooting misconfiguration issues.

Consider adding a `logInfo` call when the policy is applied.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala (hunk 1 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala:119]: **🔴 Blocking** · Correctness: Silent Data Loss `[Already posted by @Copilot — Unresolved]`

**Management SDK path silently drops `vectorIndexes` from indexing policy, making vector search non-functional**

The `getIndexingPolicy()` method (lines ~240-270) only maps `includedPaths`, `excludedPaths`, and `indexingMode` from the data-plane `IndexingPolicy` to the management SDK `IndexingPolicy`. It **silently drops** `vectorIndexes`, `compositeIndexes`, and `spatialIndexes`.

When a user sets both `vectorEmbeddingPolicy` and `indexingPolicy` with `vectorIndexes` through the management SDK path (AAD auth), the container will have the embedding policy but **no vector indexes** — making vector search non-functional. This is a pre-existing bug, but this PR makes it user-visible by enabling vector embedding policy configuration.

Copilot flagged this same issue; no response from the author yet. Consider extending `getIndexingPolicy()` to also map vector indexes.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala (hunk 2 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala:487]: **🟡 Recommendation** · Reliability: Test Coverage

**No test coverage for the management SDK client path**

This integration test only exercises the **Cosmos SDK client** path. The management SDK client (`CosmosCatalogManagementSDKClient`) has fundamentally different serialization logic — it uses `objectMapper.readValue(json, classOf[MgmtVectorEmbeddingPolicy])` with the `JsonSerializableDatabindModule` bridge, a different deserialization path from the data-plane Jackson `@JsonProperty` annotations.

Without a test, there is no proof the user-provided JSON correctly deserializes into `com.azure.resourcemanager.cosmos.models.VectorEmbeddingPolicy` via the Jackson→`JsonSerializable` bridge. The management SDK path is used for AAD auth scenarios.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala (hunk 3 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala]: **🟡 Recommendation** · Maintainability: Inconsistent Patterns

**Serialization pattern diverges from the Cosmos SDK client**

The Cosmos SDK client delegates VEP serialization to `SparkModelBridgeInternal.createVectorEmbeddingPolicyFromJson()` and `vectorEmbeddingPolicyToJson()`, centralizing serialization. This management client does inline `objectMapper.readValue()` directly.

This asymmetry means:
- Future JSON format changes need to be updated in two places
- The two paths could produce subtly different JSON during readback (`generateTblProperties`)
- The `objectMapper` here includes `JsonSerializableDatabindModule` while `SparkModelBridgeInternal` uses `Utils.getSimpleObjectMapper` — different serialization capabilities

Consider adding bridge methods for the management type, or documenting why the asymmetry is intentional.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala (hunk 4 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala:161]: **🟡 Recommendation** · Reliability: Test Gap

**Existing tests updated to size 8 but missing `VectorEmbeddingPolicy` assertion**

All six existing tests were updated from `have size 7` to `have size 8`, but none add an assertion like:

```scala
tblProperties("VectorEmbeddingPolicy") shouldEqual "null"
```

This is inconsistent with how these same tests already assert `tblProperties("AnalyticalStoreTtlInSeconds") shouldEqual "null"` for containers without analytical store TTL. Without this assertion, a bug in the null-path (e.g., returning an empty JSON object instead of `"null"`) would go undetected.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosContainerProperties.scala (hunk 5 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosContainerProperties.scala:50]: **🟡 Recommendation** · Reliability: Missing Cross-Validation

**No validation that `vectorEmbeddingPolicy` and `indexingPolicy` are consistent**

Vector search requires coordination between two properties: `vectorEmbeddingPolicy` defines embedding paths/dimensions, and `indexingPolicy.vectorIndexes` defines how those paths are indexed.

A user could provide `vectorEmbeddingPolicy` without corresponding `vectorIndexes` in `indexingPolicy` (or vice versa), or with mismatched paths. This silently creates a non-functional vector search configuration.

At minimum, consider logging a warning when `vectorEmbeddingPolicy` is provided but no `vectorIndexes` are found in `indexingPolicy`.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/models/SparkModelBridgeInternal.scala (hunk 6 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/models/SparkModelBridgeInternal.scala:19]: **🟡 Recommendation** · Reliability: Error Handling

**Malformed JSON produces a cryptic Jackson stack trace**

`objectMapper.readValue(json, classOf[CosmosVectorEmbeddingPolicy])` will throw a raw `JsonProcessingException` if the user provides malformed JSON in `TBLPROPERTIES`. Spark SQL users will see a cryptic Jackson deserialization error rather than an actionable message.

The same issue applies to the management client path (inline `objectMapper.readValue`). Consider wrapping the deserialization with a try-catch that re-throws as an `IllegalArgumentException` with a user-friendly message explaining the expected JSON format.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala (hunk 7 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala]: **🟢 Suggestion** · Reliability: Test Quality

**Substring checks could pass with broken JSON structure**

These `.contains()` assertions would pass even if the JSON structure is malformed (e.g., the key `"path"` is missing but `"vector1"` appears elsewhere in the string).

A stronger assertion would parse the JSON and check structured fields, matching the approach used above for the container properties validation (which checks `embedding.getPath`, `embedding.getDataType`, etc.):

```scala
val vepJson = tblProperties("VectorEmbeddingPolicy")
val vep = SparkModelBridgeInternal.createVectorEmbeddingPolicyFromJson(vepJson)
vep.getVectorEmbeddings should have size 1
vep.getVectorEmbeddings.get(0).getPath shouldEqual "/vector1"
```

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala (hunk 8 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala]: **🟢 Suggestion** · Correctness: Edge Case

**Non-null but empty `CosmosVectorEmbeddingPolicy` could serialize inconsistently**

If `getVectorEmbeddingPolicy` returns a non-null `CosmosVectorEmbeddingPolicy` with an empty `vectorEmbeddings` list, this serializes as `{"vectorEmbeddings":[]}` rather than `"null"`, inconsistent with other properties that show `"null"` when not meaningfully set.

Consider guarding against empty policies:

```scala
val vectorEmbeddingPolicySnapshot = Option.apply(containerProperties.getVectorEmbeddingPolicy)
    .filter(p => p.getVectorEmbeddings != null && !p.getVectorEmbeddings.isEmpty) match {
        case Some(policy) => SparkModelBridgeInternal.vectorEmbeddingPolicyToJson(policy)
        case None => "null"
    }
```

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala (hunk 9 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala:102]: **🟢 Suggestion** · Maintainability: Observability

**No logging when vector embedding policy is applied**

Neither SDK client logs when the vector embedding policy is applied during container creation. For a new feature that requires coordination with indexing policy (`vectorIndexes`), a diagnostic log message would help with troubleshooting misconfiguration issues.

Consider adding a `logInfo` call when the policy is applied.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITestBase.scala:487]: this will need to test in e2e flow.

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala (hunk 3 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala]: will consider for future improvement, keep as it is now.

File: sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosContainerProperties.scala (hunk 5 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/catalog/CosmosContainerProperties.scala:50]: this should be validated from server side.
